CMakeLists.txt

```diff
@@ -43,11 +45,21 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
 #----------------------------------------------------------------------
-# Find python3 for the build system
+# Set up virtual environment for the build system
+# This is only needed for local builds - GitHub builds install the
+# necessary dependencies in the .yml files.
 #----------------------------------------------------------------------
-find_package(Python3 COMPONENTS Interpreter)
-if(NOT ${Python3_FOUND})
-  message(WARNING "Python3 not found. Python3 is required for code generation.")
+if (NOT IS_GITHUB_BUILD)
```

Comment:
Why not do this with a github build? It would be nice to keep the process consistent and to make sure we're validating it on both OSs.